### PR TITLE
New resource and the tag API proposed by TIM.

### PR DESCRIPTION
**Pull request by pcdl1970**
New resource and the tag API proposed by TIM according comments and indications from API work group.

### CODE REVIEW

File: code/API_definitions/CAMARA/number_verification.yaml [hunk 1]
```diff
@@ -158,6 +215,21 @@ components:
           description: Hashed phone number. SHA-256 (in hexadecimal representation) of the mobile phone number in 'E164 with +' format
           type: string
           example: 32f67ab4e4312618b09cd23ed8ce41b13e095fe52b73b2e8da8ef49830e50dba
+    ueId:
+      type: object
+      properties:
+        msisdn:
```
Comment (monamok) [inline code/API_definitions/CAMARA/number_verification.yaml:221]: @bigludo7 having in mind that in this API we haven't used MSISDN, do you think we should change it to phone_number to be aligned with the other endpoints or we keep it this way?

Comment (monamok) [reply code/API_definitions/CAMARA/number_verification.yaml:221]: Also, phone_number must be required, don't you think so?

File: code/API_definitions/CAMARA/number_verification.yaml [hunk 2]
```diff
@@ -132,11 +132,68 @@ paths:
...
         - three_legged:
           - number-verification-share-read
+  /verify-ip:     
+    post:
+      tags:
+        - IP PhoneNumber Verify
+      summary: Validates that the MSISDN of the mobile device connected via the cellular network is using the IP address of the connection to the server of the service that is querying the API.
+      description: |-
+        The Mobile User inform MSISDN to verification.
+         - BackEnd APP or Web Partner Application Server pass the IP address to TIM API GTW to challenge and validate MSISDN with IP matched.
```
[the remark this answers is not in the document]
Comment (pcdl1970) [reply code/API_definitions/CAMARA/number_verification.yaml:142]: @monamok thanks for suggestion. I updated to "MNO" to be a general term.

**Comment by pcdl1970** [comment]
Dear @monamok, thanks again for your comments, suggestions and reviews. Could you, please check if you agree with new YAML in this commit https://github.com/camaraproject/NumberVerification/pull/20/commits/03147b7d01c23f93f9a91fc034dc7706ca5b0cf7

Best regards,

Paulo Lima.

**Comment by pcdl1970** [comment]
Dear @DT-DawidWroblewski , @bigludo7 could you please review API YAML commit (https://github.com/camaraproject/NumberVerification/pull/20/commits/03147b7d01c23f93f9a91fc034dc7706ca5b0cf7) regarding new resource and the tag  proposed by TIM according Camara' API group comments and suggestions.

Thanks again.

BR,

Paulo Lima.

File: code/API_definitions/CAMARA/number_verification.yaml [hunk 3]
```diff
@@ -158,6 +213,23 @@ components:
           description: Hashed phone number. SHA-256 (in hexadecimal representation) of the mobile phone number in 'E164 with +' format
           type: string
           example: 32f67ab4e4312618b09cd23ed8ce41b13e095fe52b73b2e8da8ef49830e50dba
+    ueId:
```
Comment (monamok) [inline code/API_definitions/CAMARA/number_verification.yaml:216]: About the parameters name (ipv4Addr, ipv6Addr, uePort), they are voting in other APIs working groups. We'll align them with the result that comes up from there.

File: code/API_definitions/CAMARA/number_verification.yaml [hunk 4]
```diff
@@ -158,6 +213,23 @@ components:
...
+      type: object
+      required:
+        - phone_number
+      properties:
+        phone_number:
+          description: Mobile phone number (MSISDN) to verify. The number must be in international format.
```
Comment (monamok) [inline code/API_definitions/CAMARA/number_verification.yaml:222]: Here you can reuse:
$ref: '#/components/schemas/DevicePhoneNumber'

but it's up to you.

**Review by monamok — COMMENTED** [review]
Hello @pcdl1970. Thank you for the changes and sorry for the delay. 
We merged antoher PR to align errors to CAMARA guidelines (the status changed to be an integer) and some more fixes, so now your branch is outdated. Please update it with the main branch so you can see the changes that were made. 
I added some comments. Some of them are my suggestions and we can discuss them if you don't agree. Thanks.

File: code/API_definitions/CAMARA/number_verification.yaml [hunk 5]
```diff
@@ -132,11 +134,64 @@ paths:
...
+        - oAuth2ClientCredentials:
+          - verify-ip        
 components:
   securitySchemes:
     three_legged:
       type: openIdConnect
       openIdConnectUrl: https://example.com/.well-known/openid-configuration
+    oAuth2ClientCredentials:
+          type: oauth2
+          description: Client Credentials – Intended for the server-to-server authentication between service provider and MNO.
```
Comment (monamok) [inline code/API_definitions/CAMARA/number_verification.yaml:190]:
```suggestion
          description: Client Credentials - Intended for the server-to-server authentication between service provider and MNO.
```
I think here the sign that was used is not a dash.

**Comment by pcdl1970** [comment]
Dear @monamok thanks again for your valuable comments and contributions.  And sorry for my long delay.

I updated my branch with the main branch and reviewed according your comments and contributions.

I also created a new PR https://github.com/camaraproject/NumberVerification/pull/25 regarding these updates and reviews and closed the previous PR.

Please let me know if you agree with theses changes and if it ok to request also to @DT-DawidWroblewski   and/or @bigludo7 approval to merge into main branch.

Best regards,

Paulo Lima.

**Comment by monamok** [comment]
Hello @pcdl1970 please update this branch with the chages you want to include. It's very had to follow the changes jumping from one PR to another new PR everytime we want to update something. It's not practical and it's confusing. Please modify the branch you are using for this PR to continue with it and close the new PR you opened. Thanks.

**Comment by monamok** [comment]
Dear @pcdl1970 I took a look to your new PR but please check the differences again before bringing the changes here because it seems you are overriding some of the changes that we introduced in the main branch such as for example status code which have to be integers, or one of the endpoints operationId. Thanks.

**Pull request closed without merging**